ifrs17-template/Test/IfrsVariablesTest.ipynb

```diff
@@ -181,24 +179,56 @@
         {
             "cell_type": "code",
             "source": [
-                "public List<BenchmarkTestResult> CompareAgainstBm(List<BenchmarkTestResult> errors, IEnumerable<IfrsVariable> expected, IEnumerable<IfrsVariable> computed)",
-                "\n{",
-                "\n    var expectedNotComputed = expected.Where(x => Math.Abs(x.Value) > BenchmarkPrecision).Except(computed, EqualityComparer<IfrsVariable>.Instance);",
-                "\n    var computedNotExpected = computed.Where(x => Math.Abs(x.Value) > BenchmarkPrecision).Except(expected, EqualityComparer<IfrsVariable>.Instance);",
-                "\n    if (expectedNotComputed.Any())",
-                "\n        foreach (var variable in expectedNotComputed)",
-                "\n            errors.Add(new BenchmarkTestResult(\"Extra expected variable for: Partition \" + variable.Partition + \", \" + variable.ToIdentityString()));",
-                "\n    if (computedNotExpected.Any())",
-                "\n        foreach (var variable in computedNotExpected)",
-                "\n            errors.Add(new BenchmarkTestResult(\"Extra computed variable for: Partition \" + variable.Partition + \", \" + variable.ToIdentityString()));",
+                "var includingProperties = typeof(IfrsVariable).GetIdentityProperties().Select(x=>x.Name).ToArray();"
+            ],
+            "metadata": {},
+            "execution_count": 0,
+            "outputs": []
+        },
+        {
+            "cell_type": "code",
+            "source": [
+                "using System.Reflection;",
                 "\n",
+                "\npublic List<BenchmarkTestResult> CompareIfrsVariables(IfrsVariable var1, IfrsVariable var2){",
+                "\n    var ret = new List<BenchmarkTestResult>();",
+                "\n    var values = Enumerable.Range(0, 12).Select(x => x == 0 ? \"Value\" : \"Value\" + x.ToString());",
+                "\n    foreach (var v in values){",
+                "\n        var val1 = (double)var1.GetType().GetProperty(v).GetValue(var1, null);",
+                "\n        var val2 = (double)var2.GetType().GetProperty(v).GetValue(var2, null);",
+                "\n        if (val1 + val2 > BenchmarkPrecision) ret.Add(new BenchmarkTestResult(\"Values do not match. Partition \" + var1.Partition + \", \" + var1.ToIdentityString(),  ",
+                "\n                                                                                v, val1, -val2));",
+                "\n    }",
+                "\n    return ret;",
+                "\n}"
+            ],
+            "metadata": {},
+            "execution_count": 0,
+            "outputs": []
+        },
+        {
+            "cell_type": "code",
+            "source": [
+                "public List<BenchmarkTestResult> CompareAgainstBm(List<BenchmarkTestResult> errors, IEnumerable<IfrsVariable> expected, IEnumerable<IfrsVariable> computed){",
```

Comment:
I have practically rewrote the test here, because I did not understand what the previous version was doing and it was obviously buggy. Please go through this routine and tell me if it can be done more efficiently. It written in quite a naive way, but it is at least traceable. If you think that it demands no improvement, please just close this conversation.

FYI: @dcolleoni @AlekseiNekrasov

Reply:
1. Let's use the equality comparer of IfrsVar as much as possible and reduce the use of reflection. 
2. Use ToIdentityString() instead of ToStringWith()
3. In method  CompareIfrsVariables call inputs **expected** and **computed** to make clearer the rest of the code and the log of the error message.